central/reports/scheduler/v2/report_gen.go

```diff
@@ -0,0 +1,326 @@
+package v2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/reports/common"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/mathutil"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/templates"
+	"github.com/stackrox/rox/pkg/timestamp"
+)
+
+func (s *scheduler) generateAndSendReport(req *ReportRequest, collection *storage.ResourceCollection) error {
```

Comment:
nit:  I think I would call it `generateAndSendReportResult` or something to differentiate from the config.  But some of that may be from a review of nothing but things named with `report` in them.